### PR TITLE
feat: format addDts content

### LINKED ISSUE #66: file factory (name pending) util and unindent util
Just started using this to create virtual modules and dts files. This is just an idea, but would make it easier.

```ts
// file-factory.ts
import { unindent } from './unindent'

export const fileFactory = () => {
	let file = ``

	return {
		addLines(lines: string) {
			file += unindent`${ lines }`
		},
		text() {
			return file
		}
	}
}
```

```ts
// unindent-ts
export function unindent(strings, ...values) {
    const result = [];
    const template = strings.reduce((prev, current) => prev + values.shift() + current);

    // Find the common indentation
    const indentMatch = template.match(/^\n?(\s*)/);
    const commonIndent = indentMatch ? indentMatch[1].length : 0;

    // Remove common indentation from each line
    const lines = template.split('\n');
    for (let line of lines) {
        result.push(line.slice(commonIndent));
    }

    // Join the lines back together
    return result.join('\n');
}

```

Basically means you can create virtual modules and dts files much nicer:

```ts
"astro:setup:config": () => {
  const virtualModule = fileFactory()

  virtualModule.addLines(`
    export default ${ JSON.stringify({}) }
  `)
}
```

outputs

```
export default {}
```

all nicely formatted n stuff so you don't have to worry about indents and messy af dts files if you're trying to dynamically generate

### PR DESCRIPTION
**Pull request by florian-lefebvre**
- Closes #66
- I also made unrelated improvements to the documentation, which is actually the biggest part of the PR

### CODE REVIEW

File: docs/src/content/docs/utilities/add-dts.mdx [hunk 1]
```diff
@@ -53,3 +53,7 @@ create a file inside `.astro` and reference it from `src/env.d.ts`. For example:
     ```
     </TabItem>
 </Tabs>
+
+:::note
```
Comment (Fryuni) [inline docs/src/content/docs/utilities/add-dts.mdx:57]: I think this could be a `:::tip`

**Review by Fryuni — COMMENTED** [review]
Code LGTM. Some points on the docs

**Comment by BryceRussell** [comment]
Works great, I played around with it a bunch and it generates nice looking types

**Suggestion for docs**: Is it worth documenting how to dynamically generate types? Now that you don't have to manage whitespace on your own it makes it a lot easier to generate dynamic types using a simple string buffer:

```ts
let buffer = ""

for() {
  buffer +=`\n...`
}

addDts({
    name: "my-integration",
    content: `
      declare module "virtual:my-integration" {
        ${buffer}
      }
    `
})
```

File: docs/src/content/docs/getting-started/breaking-changes.mdx
[the remark this answers is not in the document]
Comment (florian-lefebvre) [reply docs/src/content/docs/getting-started/breaking-changes.mdx:1]: This is exactly what I've done! Actually I only forgot to add redirects (or do we care?)

[the remark this answers is not in the document]
Comment (Fryuni) [reply docs/src/content/docs/getting-started/breaking-changes.mdx:1]: Riiiight! For some reason, I though it was still pointing to a different page.

**Comment by florian-lefebvre** [comment]
I added a new section related to your comment 🚀

**Review by BryceRussell — APPROVED** [review]
LGTM! 🚀

**Pull request merged**